### PR TITLE
Update compiler, tests and examples to use Svelte 5

### LINKED ISSUE #232: Migrate error and warning codes for Svelte 5 support (including in tests)
**Describe the bug**

This is related my comment in #230 about finding out that we likely need to upgrade the compiler + tests to svelte 5. It's good to know that the error + warning codes have changed the separator from `-` to `_`.

More details:
https://svelte-5-preview.vercel.app/docs/breaking-changes#other-breaking-changes-error-warning-codes-have-been-renamed

### PR DESCRIPTION
**Pull request by Greenheart**
This is based on findings in https://github.com/EMH333/esbuild-svelte/pull/230#pullrequestreview-2064452383.

Updating the main devDependencies in `package.json` to Svelte 5 and re-building the plugin was what I needed to get my app to build properly with Svelte 5.

Also updated tests to work for Svelte 5, as well as type errors in the plugin (due to changed imports and compiler options) and finally updated examples to show usage with Svelte 5 (since mounting a Svelte component will be a breaking change compared to Svelte 4).

fixes #232

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -202,7 +201,7 @@ export default function sveltePlugin(options?: esbuildSvelteOptions): Plugin {
                 dependencyModifcationTimes.set(args.path, statSync(args.path).mtime); // add the target file
 
                 let compilerOptions = {
-                    css: (svelteVersion < 3 ? false : "external") as boolean | "external",
+                    css: (svelteVersion < 3 ? undefined : "external") as CompileOptions["css"],
```
Comment (EMH333) [inline index.ts:204]: After looking at this a bit more, I'm fairly sure this logic is broken for everything between Svelte 3.0.0 and 3.53.0 (when support for "external" was added). Fun.

**Comment by EMH333** [comment]
Whoops! Could you reopen or create a new PR pointing at `main` please? I definitely want to use this as a start for the changes coming in `v1.0.0`

**Comment by zazer1** [comment]
Hi @EMH333 @Greenheart , what's the status on this? An

**Comment by EMH333** [comment]
@zazer1 please see #233, and note the current version of esbuild-svelte is compatible with Svelte 5, minus support for `.svelte.(js|ts)` files which will be coming in the next release. I know there is a lot of excitement around Svelte 5 and am working to provide full support, but also want to balance API stability and correctness

**Pull request closed without merging**